Email verification in firebase

ai-n0808 opened the pull request:
@DanielWilliamParsons 

**What I'm trying to do**
- Implement email verification. Basically the user supposed to receive email when the user registers our app. Then the user gets an email with a special verification link. Firebase knows the user's email as verified. To implement this, I added `await sendEmailVerification(user);` in `AuthContext.tsx`.

**What's the issue**
- You don't get any email even after you register it.
- It still doesn't show up the user in Firebase console after you fill up the register form and click register button.

It could be related these issues since we didn't solve this property.(https://github.com/CupCakesNSpan/exappfront/pull/55) Like why we don't see the users in Firebase console.

I'll try to figure this out as much as I can and please let me know if you have any thought.

DanielWilliamParsons commented:
@ai-n0808 
Thanks so much for doing this. I promise I will get back on to it this evening. I'll take a deep dive into the original problem. Thanks for waiting for me!

ai-n0808 commented:
@DanielWilliamParsons 
Ok! that's totally fine! No rush and pressure.
Now I'm taking look at what it could be the issue. So I'll let you know if I find something!

ai-n0808 commented:
@DanielWilliamParsons 
Actually I found something. I was wondering it might be possible `router.push("/login")` doesn't work correctly. So I tried to remove:
```
            <Link href="/login" className={styles.redirectLink}>
              Go to login page
            </Link>
```
Then basic form validation is working! (I believe it didn't work before) It shows up error message like "All fields are required." if I try to submit with empty input.
 
But instead I'm getting this error
`Firebase: Error (auth/api-key-not-valid.-please-pass-a-valid-api-key.)`
<img width="380" alt="Screenshot 2025-02-21 at 0 22 30 PM" src="https://github.com/user-attachments/assets/9601028a-98e3-4d90-a975-4587afb083a8" />

Overall what might be the causes are
- `router.push("/login")` is not working property.
- This firebase error.

DanielWilliamParsons commented:
@ai-n0808 
Thanks for keeping this going.
Sorry I have been so slow. I got sick last night, and I'm still a little sick today with a fever.
I need a little more rest, and then I'll be back onto it.

I'll check out the firebase error. As for the router.push - yes that makes sense. I'll need to have a closer look to see why validation doesn't work when router.push is made available.

Again, sorry. I'm being really slow. I'll get back to it soon...😷

ai-n0808 commented:
@DanielWilliamParsons 
Oh no.. sorry to hear that. No need to apologize Daniel. That's totally fine and pls take care😭
Ok, thank you!! 
And sorry I put codes above, it's wrong. I was gonna paste this code instead of above.(It's around Register button. Not login)
```
          {/* <Link href="/login"> */}
            <button type="submit" className={styles.registerButton}>
              Register
            </button>
          {/* </Link> */}
```

Again, no worries! Get some rest and take your time pls🙏🏻

ai-n0808 commented:
@DanielWilliamParsons 
I pushed my code after I cleaned up and modified everything.
So feel free to take a look and lmk if there's anything else. No rush :))
